### PR TITLE
Feature/sc 109542/ability to move a folder

### PR DESCRIPTION
**Pull request by markhornak**
**Overview**
Shortcut: [109542](https://app.shortcut.com/guru/story/109542/ability-to-move-a-folder)

Adds the ability to move a folder to another folder in the collection.  Additional helper methods were added to support Parent Folder, Homer Folder and delete() from the Folder object.

- `get_parent_folder`, `get_home_folder`. `remove_card_from_folder` in `core.py`
- `get_parent`,`get_home`, `remove_card`, `delete`  in `data_objects.py`

### CODE REVIEW

File: guru/core.py [hunk 1]
```diff
@@ -2501,6 +2501,58 @@ def remove_card_from_folder(self, card, folder):
...
+            }
+        ],
+        "prevSiblingItemId": "first"
+    }
+
+    # clear the cache for the folder since we moved a card...
+    self.__clear_cache(
+        f"{self.base_url}/folders/{target_folder_obj.id}/items")
+    self.__clear_cache(
+        f"{self.base_url}/folders/{source_folder_slug}/items")
```
Comment (jahlbornG) [inline guru/core.py:2545]: the comment isn't accurate, and you don't need to clear the source folder items.

File: guru/core.py [hunk 2]
```diff
@@ -2501,6 +2501,58 @@ def remove_card_from_folder(self, card, folder):
...
+
+    url = f"{self.base_url}/folders/{target_folder_obj.id}/action"
+    response = self.__post(url, data)
+    if status_to_bool(response.status_code):
+      # refresh the internal items for the source and target Folders
+      source_folder_obj.update_lists(source_folder_obj, "remove")
+      target_folder_obj.update_lists(target_folder_obj, "add")
```
Comment (jahlbornG) [inline guru/core.py:2552]: you don't need to do anything to the source folder.  the target folder should be adding the source_folder_obj.

File: guru/core.py (hunk 1 above)
Comment (markhornak) [reply guru/core.py:2545]: ah, right!

File: guru/core.py (hunk 2 above)
Comment (markhornak) [reply guru/core.py:2552]: Yup, same as above...

File: guru/core.py (hunk 1 above)
Comment (markhornak) [reply guru/core.py]: Fixed here: [4490123](https://github.com/guruhq/py-sdk/pull/60/commits/4490123664b8dede56cf8fb2fa8e8f5ec1eb006f)

File: guru/core.py (hunk 2 above)
Comment (markhornak) [reply guru/core.py]: Fixed here: [4490123](https://github.com/guruhq/py-sdk/pull/60/commits/4490123664b8dede56cf8fb2fa8e8f5ec1eb006f)

**Review by jahlbornG — APPROVED** [review]

**Pull request merged**